server/src/main/java/org/apache/cassandra/sidecar/handlers/InvalidateCacheHandler.java

```diff
@@ -78,34 +80,85 @@ public Set<Authorization> requiredAuthorizations()
     }
 
     @Override
-    protected String extractParamsOrThrow(RoutingContext context)
+    protected Params extractParamsOrThrow(RoutingContext context)
     {
-        return context.pathParam("cacheName");
+        String cacheName = context.pathParam("cacheName");
+        List<String> keys = context.queryParam("keys");
+        return new Params(cacheName, keys);
+    }
+
+    /**
+     * Simple holder class for cache invalidation parameters
+     */
+    protected static class Params
+    {
+        final String cacheName;
+        final List<String> keys;
+
+        Params(String cacheName, List<String> keys)
+        {
+            this.cacheName = cacheName;
+            this.keys = keys;
+        }
+
+        boolean invalidateAll()
+        {
+            return keys == null || keys.isEmpty();
+        }
     }
 
     @Override
     protected void handleInternal(RoutingContext context,
                                   HttpServerRequest httpRequest,
                                   @NotNull String host,
                                   SocketAddress remoteAddress,
-                                  String cacheName)
+                                  Params params)
     {
+        String cacheName = params.cacheName;
+        boolean invalidateAll = params.invalidateAll();
+
         switch (cacheName.toLowerCase())
         {
             case "identity_to_role_cache":
```

Comment:
Can you reuse cache `NAME` here https://github.com/apache/cassandra-sidecar/blob/f41b3ca8cc8b3f89fb5f5fa5ba760c42fd4df8e0/server/src/main/java/org/apache/cassandra/sidecar/acl/IdentityToRoleCache.java#L37. Also can you add static constants for names without underscore? incase we want to reuse name.